src/deadline/client/ui/dialogs/submit_job_to_deadline_dialog.py

```diff
@@ -400,6 +411,18 @@ def on_submit(self):
 
         # Submit the job
         try:
+            # Execute any PreSubmission function defined.
+            if self.on_pre_submit_callback:
+                self.on_pre_submit_callback(
+                    self,
+                    job_history_bundle_dir,
+                    settings,
+                    queue_parameters,
+                    asset_references,
+                    purpose=JobBundlePurpose.SUBMISSION
```

Comment:
I believe the intent here is to match what we pass into `on_create_job_bundle_callback` to give users full control before the job gets created. So we should be including requirements here as well.

Reply:
You're correct about that, I grabbed the on_create_job_bundle_callback signature from the else block.

I can move it below where `requirements` is defined.